composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/MySpotsView.kt

```diff
@@ -31,170 +31,213 @@ import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
+import com.overdrive.cruiser.models.BookingsViewModel
 import com.overdrive.cruiser.models.MySpotsViewModel
+import com.overdrive.cruiser.models.Spot
 import cruiser.composeapp.generated.resources.Res
 import cruiser.composeapp.generated.resources.accessibility_on
 import cruiser.composeapp.generated.resources.weather_on
 import kotlinx.coroutines.launch
+import kotlinx.datetime.Clock
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
 import org.jetbrains.compose.resources.vectorResource
 
 @Composable
-fun MySpotsView(mySpotsViewModel: MySpotsViewModel, onAddSpotClick: () -> Unit) {
-
-    val spots by mySpotsViewModel.spots.collectAsState()
-    val scope = rememberCoroutineScope()
-    var showDialog by remember { mutableStateOf(false) }
-    val spot = spots.firstOrNull { it.ownerId == "dev3" }
+fun MySpotsView(mySpotsViewModel: MySpotsViewModel, bookingsViewModel: BookingsViewModel,
+                onAddSpotClick: () -> Unit) {
+    var selectedSpot by remember { mutableStateOf<Spot?>(null) }
 
     Column {
-        SpotOnTopBar("My Spot")
-        LaunchedEffect(Unit) {
-            mySpotsViewModel.updateSpots()
+        if (selectedSpot != null) {
+            MySpotInformation(mySpotsViewModel, selectedSpot!!, onAddSpotClick) { selectedSpot = null }
+        } else {
+            MySpotsList(mySpotsViewModel, bookingsViewModel, onAddSpotClick, onSpotSelected = {
+                selectedSpot = it
+            })
         }
+    }
+}
 
-        Box(modifier = Modifier.fillMaxSize()) {
+@Composable
+fun MySpotsList(mySpotsViewModel: MySpotsViewModel, bookingsViewModel: BookingsViewModel,
+                onAddSpotClick: () -> Unit, onSpotSelected: (Spot) -> Unit) {
+    val allSpots by mySpotsViewModel.spots.collectAsState()
+    val devSpots = allSpots.filter { it.ownerId == "dev3" }
```

Comment:
I'm starting to worry we are gonna lose track of all these usages, but I suppose we'll just pray 🙏 .